src/main/java/it/pagopa/pn/delivery/svc/search/NotificationRetrieverService.java

```diff
@@ -550,4 +573,42 @@ private void notifyNotificationViewedEvent(InternalNotification notification, St
 		Instant createdAt = clock.instant();
 		notificationAcknowledgementProducer.sendNotificationViewed( iun, createdAt, recipientIndex );
 	}
+
+	private void labelizeGroup(InternalNotification notification, String senderId) {
+		String notificationGroup = notification.getGroup();
+		if (notificationGroup != null && !notificationGroup.isEmpty() && senderId != null) {
+			List<PaGroup> groups = pnExternalRegistriesClient.getGroups(senderId);
+			if (groups != null) {
+				PaGroup group = groups.stream()
+						.filter(g -> g.getId().equals(notificationGroup))
+						.findAny()
+						.orElse(null);
+				if (group != null) {
+					notification.setGroup(group.getName());
+				}
+			}
+		}
+	}
+
+	private void labelizeGroups(ResultPaginationDto<NotificationSearchRow,PnLastEvaluatedKey> searchResult, String senderId) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.

Reply:
Complessità ridotta togliendo if statement annidati